File: backend/src/main/java/space/davids_digital/vk_gpt_bot/service/VkAuthService.java
package space.davids_digital.vk_gpt_bot.service;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import space.davids_digital.vk_gpt_bot.integration.vk.VkApiService;
import space.davids_digital.vk_gpt_bot.model.UserSessionModel;
import space.davids_digital.vk_gpt_bot.orm.service.UserSessionOrmService;

import java.time.ZonedDateTime;
import java.util.UUID;

@Service
public class VkAuthService {
    private static final Logger log = LoggerFactory.getLogger(VkAuthService.class);

    private final VkApiService vkApiService;
    private final UserSessionOrmService userSessionOrmService;
    private final SessionService sessionService;

    @Autowired
    public VkAuthService(
            VkApiService vkApiService,
            UserSessionOrmService userSessionOrmService,
            SessionService sessionService
    ) {
        this.vkApiService = vkApiService;
        this.userSessionOrmService = userSessionOrmService;
        this.sessionService = sessionService;
    }

    public UserSessionModel createSessionFromSilentToken(String silentToken, UUID uuid) {
        log.info("Creating session from silent token");
        var result = vkApiService.exchangeSilentTokenToAccessToken(silentToken, uuid);
        var sessionToken = sessionService.createNewSessionToken();
        return userSessionOrmService.createUserSession(
                result.userId,
                sessionToken,
                result.accessToken,
                result.accessTokenId,
                ZonedDateTime.now().plusSeconds(result.expiresIn)
        );
    }
}

File: backend/src/main/java/space/davids_digital/sweetie/integration/vk/VkAuthService.kt
package space.davids_digital.sweetie.integration.vk

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import space.davids_digital.sweetie.model.UserSessionModel
import space.davids_digital.sweetie.orm.service.UserSessionOrmService
import space.davids_digital.sweetie.service.SessionService
import java.time.ZonedDateTime
import java.util.*

@Service
class VkAuthService @Autowired constructor(
    private val vkRestApiService: VkRestApiService,
    private val userSessionOrmService: UserSessionOrmService,
    private val sessionService: SessionService
) {
    companion object {
        private val log = LoggerFactory.getLogger(VkAuthService::class.java)
    }

    fun createSessionFromSilentToken(silentToken: String, uuid: UUID): UserSessionModel {
        log.info("Creating session from silent token")
        val result = vkRestApiService.exchangeSilentTokenToAccessToken(silentToken, uuid)
        val sessionToken = sessionService.createNewSessionToken()
        return userSessionOrmService.createUserSession(
            result.userId,
            sessionToken,
            result.accessToken!!,
            result.accessTokenId!!,
            ZonedDateTime.now().plusSeconds(result.expiresIn)
        )
    }
}
